### PR TITLE
Issue 81- ResourceResolver should be closed in try-with-resources

### LINKED ISSUE #81: ResourceResolver + autoclosable

### PR DESCRIPTION
**Pull request by GulderBone**
fixes #81

### CODE REVIEW

File: src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Map;
+import java.util.List;
+import java.awt.Event;
+import jdk.nashorn.internal.ir.annotations.Reference;
+import org.apache.sling.api.SlingConstants;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+
+public class AutoclosableResourceResolver {
```
Comment (michaltobiasz) [inline src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java]: General comment:
Could you please change method names to more descriptive. I think they should describe what is tested (what case of resource resolver is tested)

File: src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    private static final String SLING_RESOURCE_RESOLVER = "org.apache.sling.api.resource.ResourceResolver";
+
+    private boolean insideTryStatement = false;
+
+    private boolean insideLambdaExpression = false;
+
+    private JavaFileScannerContext context;
+
+    private List<String> tryWithResourceResources = new ArrayList<>();
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java]: Please change name. Maybe ResourceResolversInTryWithResource ?

File: src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    protected void case1() {
+        ResourceResolver resourceResolver = null;
+        try {
+            resourceResolver = resourceResolverFactory.getServiceResourceResolver(null); // Noncompliant
+        } catch (LoginException e) {
+            System.out.println("something went wrong");
+        } finally {
+            if (resourceResolver != null) {
+                //resourceResolver.close();
```
[the remark this answers is not in the document]
Comment (GulderBone) [reply src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java]: deleted

File: src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java (hunk 2 above)
Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java]: done

File: src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java (hunk 1 above)
Comment (GulderBone) [reply src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java]: Done

File: src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    private static final String SLING_RESOURCE_RESOLVER = "org.apache.sling.api.resource.ResourceResolver";
+
+    private boolean insideTryStatement = false;
+
+    private boolean insideLambdaExpression = false;
+
+    private JavaFileScannerContext context;
+
+    private List<String> ResourceResolversInTryWithResources = new ArrayList<>();
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/checks/resourceresolver/close/ResourceResolverTryWithResourcesCheck.java]: name should not start with upper case letter

File: src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            ResourceResolver resourceResolver = resourceResolverProducer.produce(); // Noncompliant
+        } finally {
+            if (null != resourceResolver && resourceResolver.isLive()) {
+                resourceResolver.close();
+            }
+        }
+        return name;
+    }
+
+    private void resourceResolverShouldBeIgnoredInThisKindOfCase(Event event, List<String> attributesList) {
```
Comment (michaltobiasz) [inline src/test/files/checks/ResourceResolver/AutoclosableResourceResolver.java]: Did you mean that ThisKindOfCase == WhenResolverFactoryIsExecutedInMethodFromDifferentClass ?

**Comment by GulderBone** [comment]
New pull request: https://github.com/Cognifide/AEM-Rules-for-SonarQube/pull/154

**Pull request closed without merging**